### PR TITLE
ECER-1758: Move checkboxes from education list to inside education entry

### PR DESCRIPTION
**Pull request by qsl-deepak**
Move checkboxes from education list to inside education entry
---
This PR aims at moving the education list to inside education entry

## Title
ECER-{1758}: Move checkboxes from education list to inside education entry

## Description

- Expose ecer_IsECEAssistant in endpoint and map to dynamics
- Expose ecer_DoesECERegistryHaveTranscript in endpoint and map to dynamics
- Expose ecer_IsOfficialTranscriptRequested in endpoint and map to dynamics
- FE - Move checkboxes from education list to inside education entry

## Related Jira Issue(s)

- ECER-{716}
- ECER-{1757}
- ECER-{1758}

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)
Added to:
image
Validation:
image
Removed from:
image

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 1]
```diff
@@ -68,6 +68,23 @@
...
+            color="primary"
+            label="I have requested the official transcript from my education institution"
+          ></v-checkbox>
+          <v-checkbox
+            v-model="officialTranscriptReceived"
+            :class="{ 'error-message': !atLeastOneChecked }"
+            color="primary"
+            label="The ECE Registry already has my official transcript for the course/program relevant to this application and certificate type"
+          ></v-checkbox>
+          <div v-if="!atLeastOneChecked" class="v-messages error-message" role="alert">Please select at least one Option</div>
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:84]: Is this copy from figma? I think "...Option" should be lowercase

Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:84]: Should this be a rule?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 2]
```diff
@@ -262,3 +272,8 @@ export default defineComponent({
   },
 });
 </script>
+<style>
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:275]: You should be able to use the color prop and remove `.error-message` class all together

https://vuetifyjs.com/en/api/v-alert/#props-color

Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:275]: Ah sorry, this is for just displaying text in red? In that case I _think_ `text-error` would work

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue (hunk 1 above)
Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue]: updated text from figma.

Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue]: using :rule to validate. added div to toggle error message for now

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue (hunk 2 above)
Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue]: we are using :rules to enforce validation on the checkboxes, but we do not want error messages to show for both of them individually(as per figma designs) so for now we added a div that gets toggled based on the :rules validation.
Note: need to revisit this an clean up the code to toggle validation message via :rules prop

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 3]
```diff
@@ -158,7 +161,12 @@ export default defineComponent({
     newClientId() {
       return Object.keys(this.modelValue).length + 1;
     },
+
+    atLeastOneCheckedError() {
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:165]: can rename to `atLeastOneChecked` and you can add return type which is nice:

`atLeastOneChecked(): boolean {...`

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 4]
```diff
@@ -259,6 +271,12 @@ export default defineComponent({
       this.endYear = "";
     },
     formatDate,
+    atLeastOneCheckedRule() {
+      if (!this.officialTranscriptRequested && !this.officialTranscriptReceived) {
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:275]: Can we consolidate this logic, I think you should be able to do something like:

```
if (!this.atLeastOneChecked) {
...
}
```

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 5]
```diff
@@ -68,6 +68,23 @@
...
+            label="I have requested the official transcript from my education institution"
+          ></v-checkbox>
+          <v-checkbox
+            ref="refOfficialTranscriptReceived"
+            v-model="officialTranscriptReceived"
+            :rules="[atLeastOneCheckedRule]"
+            color="primary"
+            label="The ECE Registry already has my official transcript for the course/program relevant to this application and certificate type"
+          ></v-checkbox>
+          <div v-if="!atLeastOneCheckedError" class="v-messages error-message mb-5" role="alert">Indicate the status of your trnscript(s)</div>
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue]: Typo **transcript(s)**

Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue]: Also, I assume this div shows on initial load?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue [hunk 6]
```diff
@@ -68,6 +68,23 @@
           maxlength="50"
           class="my-8"
         ></v-text-field>
+        <v-row>
+          <v-checkbox
+            ref="refOfficialTranscriptRequested"
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue:73]: Let's remove these ref's if we aren't using them

**Review by pcatkins — CHANGES_REQUESTED** [review]
See comments.

**Review by pcatkins — APPROVED** [review]

**Pull request merged**